mslib/mscolab/file_manager.py

```diff
@@ -250,6 +256,59 @@ def modify_user(self, user, attribute=None, value=None, action=None):
             db.session.commit()
         return True
 
+    def upload_file(self, file, upload_folder, file_token=None, subfolder=None, identifier=None):
+        """
+        Generic function to save files securely in specified directory with unique filename
+        and return the relative file path.
+        """
+        with fs.open_fs(upload_folder):
+            file_dir = fs.path.join(upload_folder, str(subfolder) if subfolder else "")
+            if sys.platform.startswith('win'):
+                file_dir = file_dir.replace('\\', '/')
+            if not os.path.exists(file_dir):
+                os.makedirs(file_dir)
+
+            # Creating unique and secure filename
+            file_name, _ = file.filename.rsplit('.', 1)
+            file_ext = guess_extension(file.mimetype) or '.unknown'
+
+            token = file_token if file_token else secrets.token_urlsafe(32)
```

Comment:
Due to the reason described here: https://docs.python.org/3/library/secrets.html#how-many-bytes-should-tokens-use I don't think there should be an explicit number of bytes set for `token_urlsafe`. Instead, without an argument it will choose an "appropriate" default, which will be increased by the python stdlib maintainers as deemed necessary.